app/docker/containers/queries.ts

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from 'react-query';
+
+import { EnvironmentId } from '@/portainer/environments/types';
+import { error as notifyError } from '@/portainer/services/notifications';
+
+import { Filters, getContainers } from './containers.service';
+
+export function useContainers(
+  environmentId: EnvironmentId,
+  all = true,
+  filters?: Filters,
+  autoRefreshRate?: number
+) {
+  return useQuery(
+    ['docker', environmentId, 'containers', { all, filters }],
+    () => getContainers(environmentId, all, filters),
+    {
+      onError(err) {
+        notifyError('Failure', err as Error, 'Unable to retrieve containers');
+      },
+      refetchInterval() {
+        if (autoRefreshRate) {
```

Comment:
```suggestion
        return !!autoRefreshRate
```

Reply:
!!autoRefreshRate always return a boolean, while we want a `number | false`, but using `??` we can get what we need